From 9363f0583b7cd7a816d90523ed1ae61e46d37d2b Mon Sep 17 00:00:00 2001
From: Rusty Lynch <rusty.lynch@intel.com>
Date: Mon, 27 Jul 2009 16:31:32 -0700
Subject: [PATCH 1/4] CarrickNotificationManager: Fix for bug #2770

Use our stashed away network name or network type for the
first part of the message instead of the currently connected
name/type.
---
 carrick/carrick-notification-manager.c |   15 +++++++++++++--
 1 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/carrick/carrick-notification-manager.c b/carrick/carrick-notification-manager.c
index 0c13246..83e5dca 100644
--- a/carrick/carrick-notification-manager.c
+++ b/carrick/carrick-notification-manager.c
@@ -230,15 +230,26 @@ _tell_changed (CarrickNotificationManager *self,
   gchar *message = NULL;
   const gchar *icon;
 
+  if (priv->last_name == NULL && priv->last_type == NULL)
+  {
+    /*
+     * If we have never been notified of a previous network 
+     * name or network type then we it would be better to just
+     * not send a notification.
+     */
+    g_free (title);
+    return;
+  }
+
   if (priv->last_name)
   {
     old = g_strdup_printf (_("Sorry, your connection to %s was lost. So we've "),
-                           name);
+                           priv->last_name);
   }
   else
   {
     old = g_strdup_printf (_("Sorry, your %s connection was lost. So we've "),
-                           type);
+                           priv->last_type);
   }
 
   if (g_strcmp0 (type, "ethernet") == 0)
-- 
1.6.2.5

